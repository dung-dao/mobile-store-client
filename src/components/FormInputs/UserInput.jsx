import React from 'react';
import PropTypes, {number} from "prop-types";
import IF from "../common/IF";
import {Col, Form, Input, Select} from "antd";
import {numberRegex, vietnameseRegex} from "../../utils/validate";

const UserInput = (props) => {
    const readOnly = props.readOnly;
    const span = props.span;
    return (
        <React.Fragment>
            <IF condt={props.action !== 'CREATE'}>
                <Col xs={24} md={span}>
                    < Form.Item
                        label="ID"
                        name="id"
                    >
                        < Input
                            readOnly={true}
                        />
                    </Form.Item>
                </Col>
            </IF>
            <Col xs={24} md={span}>
                <Form.Item
                    label="Tên đăng nhập"
                    name="name"
                    rules={[
                        {required: true, message: 'Vui lòng nhập tên đăng nhập'},
                        {max: 255, message: 'Tên vượt quá độ dài cho phép'},
                        {pattern: vietnameseRegex, message: 'Tên đăng nhập không hợp lệ'}
                    ]}
                >
                    <Input readOnly={readOnly}/>
                </Form.Item>
            </Col>

            <Col xs={24} md={span}>
                <Form.Item
                    label="Vai trò"
                    name="role"
                    rules={[
                        {required: true, message: 'Vui lòng nhập vai trò'},
                    ]}
                >
                    {/*TODO: Thêm Role chính xác*/}
                    <Select
                        options={[
                            {label: "Admin", value: "admin"},
                            {label: "Nhân viên", value: "admin"},
                            {label: "Thủ kho", value: "admin"}
                        ]}
                    />
                </Form.Item>
            </Col>
        </React.Fragment>
    );
};

UserInput.propTypes = {
    span: PropTypes.number.isRequired,
    readOnly: PropTypes.bool.isRequired,
    action: PropTypes.string.isRequired
};


export default UserInput;