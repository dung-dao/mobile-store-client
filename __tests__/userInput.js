import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, shallow} from 'enzyme';
import UserInput from "../src/components/FormInputs/UserInput";

configure({adapter: new Adapter()});

configure({adapter: new Adapter()});
describe('User input', () => {
    it('Match snapshoot create', () => {
        const tree = shallow(<UserInput span={4} action="CREATE" readOnly={false}/>);
        expect(tree).toMatchSnapshot();
    });
    it('Match snapshoot view', () => {
        const tree = shallow(<UserInput span={4} action="VIEW" readOnly={true}/>);
        expect(tree).toMatchSnapshot();
    });
})
