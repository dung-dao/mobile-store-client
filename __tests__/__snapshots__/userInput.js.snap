// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User input Match snapshoot create 1`] = `
<Fragment>
  <IF
    condt={false}
  >
    <Col
      md={4}
      xs={24}
    >
      <FormItem
        label="ID"
        name="id"
      >
        <Input
          readOnly={true}
          type="text"
        />
      </FormItem>
    </Col>
  </IF>
  <Col
    md={4}
    xs={24}
  >
    <FormItem
      label="Tên đăng nhập"
      name="name"
      rules={
        Array [
          Object {
            "message": "Vui lòng nhập tên đăng nhập",
            "required": true,
          },
          Object {
            "max": 255,
            "message": "Tên vượt quá độ dài cho phép",
          },
          Object {
            "message": "Tên đăng nhập không hợp lệ",
            "pattern": /\\^\\[\\\\t-\\\\r A-Z_a-z\\\\xA0\\\\xC0-\\\\xC3\\\\xC8-\\\\xCA\\\\xCC\\\\xCD\\\\xD2-\\\\xD5\\\\xD9\\\\xDA\\\\xDD\\\\xE0-\\\\xE3\\\\xE8-\\\\xEA\\\\xEC\\\\xED\\\\xF2-\\\\xF5\\\\xF9\\\\xFA\\\\u0102\\\\u0103\\\\u0110\\\\u0111\\\\u0128\\\\u0129\\\\u0168\\\\u0169\\\\u01A0\\\\u01A1\\\\u01AF\\\\u01B0\\\\u1680\\\\u1EA0-\\\\u1EBD\\\\u1EC0-\\\\u1EF9\\\\u2000-\\\\u200A\\\\u2028\\\\u2029\\\\u202F\\\\u205F\\\\u3000\\\\uFEFF\\]\\+\\$/,
          },
        ]
      }
    >
      <Input
        readOnly={false}
        type="text"
      />
    </FormItem>
  </Col>
  <Col
    md={4}
    xs={24}
  >
    <FormItem
      label="Vai trò"
      name="role"
      rules={
        Array [
          Object {
            "message": "Vui lòng nhập vai trò",
            "required": true,
          },
        ]
      }
    >
      <Select
        bordered={true}
        choiceTransitionName="zoom"
        options={
          Array [
            Object {
              "label": "Admin",
              "value": "admin",
            },
            Object {
              "label": "Nhân viên",
              "value": "admin",
            },
            Object {
              "label": "Thủ kho",
              "value": "admin",
            },
          ]
        }
        transitionName="slide-up"
      />
    </FormItem>
  </Col>
</Fragment>
`;

exports[`User input Match snapshoot view 1`] = `
<Fragment>
  <IF
    condt={true}
  >
    <Col
      md={4}
      xs={24}
    >
      <FormItem
        label="ID"
        name="id"
      >
        <Input
          readOnly={true}
          type="text"
        />
      </FormItem>
    </Col>
  </IF>
  <Col
    md={4}
    xs={24}
  >
    <FormItem
      label="Tên đăng nhập"
      name="name"
      rules={
        Array [
          Object {
            "message": "Vui lòng nhập tên đăng nhập",
            "required": true,
          },
          Object {
            "max": 255,
            "message": "Tên vượt quá độ dài cho phép",
          },
          Object {
            "message": "Tên đăng nhập không hợp lệ",
            "pattern": /\\^\\[\\\\t-\\\\r A-Z_a-z\\\\xA0\\\\xC0-\\\\xC3\\\\xC8-\\\\xCA\\\\xCC\\\\xCD\\\\xD2-\\\\xD5\\\\xD9\\\\xDA\\\\xDD\\\\xE0-\\\\xE3\\\\xE8-\\\\xEA\\\\xEC\\\\xED\\\\xF2-\\\\xF5\\\\xF9\\\\xFA\\\\u0102\\\\u0103\\\\u0110\\\\u0111\\\\u0128\\\\u0129\\\\u0168\\\\u0169\\\\u01A0\\\\u01A1\\\\u01AF\\\\u01B0\\\\u1680\\\\u1EA0-\\\\u1EBD\\\\u1EC0-\\\\u1EF9\\\\u2000-\\\\u200A\\\\u2028\\\\u2029\\\\u202F\\\\u205F\\\\u3000\\\\uFEFF\\]\\+\\$/,
          },
        ]
      }
    >
      <Input
        readOnly={true}
        type="text"
      />
    </FormItem>
  </Col>
  <Col
    md={4}
    xs={24}
  >
    <FormItem
      label="Vai trò"
      name="role"
      rules={
        Array [
          Object {
            "message": "Vui lòng nhập vai trò",
            "required": true,
          },
        ]
      }
    >
      <Select
        bordered={true}
        choiceTransitionName="zoom"
        options={
          Array [
            Object {
              "label": "Admin",
              "value": "admin",
            },
            Object {
              "label": "Nhân viên",
              "value": "admin",
            },
            Object {
              "label": "Thủ kho",
              "value": "admin",
            },
          ]
        }
        transitionName="slide-up"
      />
    </FormItem>
  </Col>
</Fragment>
`;
